packages/@aws-cdk/aws-cloudtrail/lib/index.ts

```diff
@@ -181,6 +181,11 @@ export class CloudTrail extends cdk.Construct {
     this.cloudTrailArn = trail.trailArn;
     const s3BucketPolicy = s3bucket.node.findChild("Policy").node.findChild("Resource") as s3.CfnBucketPolicy;
     trail.node.addDependency(s3BucketPolicy);
+
+    if (logsRole !== undefined) {
+      const logsRolePolicy = logsRole.node.findChild("DefaultPolicy").node.findChild("Resource");
+      trail.node.addDependency(logsRolePolicy);
```

Comment:
Can you add a comment explaining why we need this?

Reply:
Don't even need to do that, you can do:

```ts
trail.node.addDependency(logsRolePolicy);
```

Reply:
I assume you mean
```typescript
trail.node.addDependency(logsRole);
```
I tried this and it works. It adds both role and the policy as dependencies.

Reply:
I'll add this as a comment:

If props.sendToCloudWatchLogs is set to true then the trail needs to depend on the created logsRole so that it can create the log stream for the log group. This ensures the logsRole is created and propagated before the trail tries to create the log stream.

Without the dependency the cloud trail fails to deploy with the same error as the referenced issue.